app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -355,34 +372,69 @@ function RawScatterPlot({
   useEffect(() => {
     setIsLoading(true)
 
+    let expressionArray
+
+    const fetchMethod = dataCache ? dataCache.fetchCluster : fetchCluster
+
     // use an image and/or data cache if one has been provided, otherwise query scp-api directly
     if (
       flags?.progressive_loading && isGeneExpression(genes, isCorrelatedScatter) && !isAnnotatedScatter &&
-      !scatterData &&
-      genes[0] === 'A1BG-AS1' // Placeholder; likely replace with setting like DE
+      !scatterData
     ) {
-      const bucketName = 'broad-singlecellportal-public'
-      const filePath = `test/scatter_image/${genes[0]}-v2.webp`
-      fetchBucketFile(bucketName, filePath).then(async response => {
-        renderImage(response)
+      const urlSafeCluster = cluster.replaceAll(' ', '_')
```

Comment:
Blocker: we will need to handle the corner case where a cluster name has `+` in it (like `CD4+`) and had that converted to `pos` when caching images/data.  Also, the logic for the replaceAll should cover non-word characters, not just spaces.  The `ExpressionWriter` class in `scp-ingest-pipeline` has the following logic:

```
plus_converted_string = name.replace('+', 'pos')
return re.sub(r'\W', '_', plus_converted_string)
```

We should also ensure that Image Pipeline is doing the same thing.  I checked and it doesn't appear that there is any cluster name sanitization happening that I can see.  None of our testing data covers this case but there are clusters in production that contain a plus character in their name.

Reply:
Good point, agreed.  I've got another PR in the works that fixes some bugs in Image Pipeline and related integration.  I'll fix this code here as you request, and make the related fix in Image Pipeline in that forthcoming PR.

Reply:
Done in 163be78.  I also just opened SCP-4846 to consolidate our 3-4 implementations of this, per chat.